### ISSUE TITLE
Matrix instantiation of Corr objects does not work when elements are None

### ISSUE THREAD

**Issue by fjosw** [fjosw/pyerrors#65]
Instantiation of a multi dimensional correlator matrix as an array of `Corr` objects results in an error if an entry of the correlators is `None`.

**Example:**
```python
import numpy as np
import pyerrors as pe

dim = 64
x = np.arange(dim)
y = 2 * np.exp(-0.06 * x) + np.random.normal(0.0, 0.15, dim)
yerr = [0.1] * dim

oy = []
for i, item in enumerate(x):
    oy.append(pe.pseudo_Obs(y[i], yerr[i], 'test'))
    
corr = pe.Corr(oy)
corr = corr.deriv()
pe.Corr(np.array([[corr, corr], [corr, corr]]))
```

**Error:**
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_19547/2202667383.py in <module>
     10 corr = pe.Corr(oy)
     11 corr = corr.deriv()
---> 12 pe.Corr(np.array([[corr, corr], [corr, corr]]))

~/Codes/pyerrors/pyerrors/correlators.py in __init__(self, data_input, padding, prange)
     56             input_as_list = []
     57             for t in range(T):
---> 58                 if any([(item.content[t][0] is None) for item in data_input.flatten()]):
     59                     if not all([(item.content[t][0] is None) for item in data_input.flatten()]):
     60                         warnings.warn("Input ill-defined at different timeslices. Conversion leads to data loss!", RuntimeWarning)

~/Codes/pyerrors/pyerrors/correlators.py in <listcomp>(.0)
     56             input_as_list = []
     57             for t in range(T):
---> 58                 if any([(item.content[t][0] is None) for item in data_input.flatten()]):
     59                     if not all([(item.content[t][0] is None) for item in data_input.flatten()]):
     60                         warnings.warn("Input ill-defined at different timeslices. Conversion leads to data loss!", RuntimeWarning)

TypeError: 'NoneType' object is not subscriptable
```

**Comment by JanNeuendorf** [comment]
I think the official term here is "ooops". I will try to fix this in the next push.

**Comment by fjosw** [comment]
Bug fixed in #76, test added in dce7d4bec825c294a184c970684dc2c19ff47e5d.
